### PR TITLE
tests for tslibs.conversion and tslibs.timezones

### PR DESCRIPTION
**Pull request by jbrockmendel**
This is the last one.

### CODE REVIEW

File: pandas/tests/tslibs/test_conversion.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            tm.assert_numpy_array_equal(result, result_single)
+
+        # US: 2014-03-09 - 2014-11-11
+        # MOSCOW: 2014-10-26  /  2014-12-31
+        tz_didx = date_range('2014-03-01', '2015-01-10', freq='H', tz=tz)
+        utc_didx = date_range('2014-03-01', '2015-01-10', freq='H')
+        compare_utc_to_local(tz_didx, utc_didx)
+        # local tz to UTC can be differ in hourly (or higher) freqs because
+        # of DST
+        compare_local_to_utc(tz_didx, utc_didx)
```
Comment (jreback) [inline pandas/tests/tslibs/test_conversion.py]: you could parametrize

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/tslibs/test_conversion.py (hunk 1 above)
Comment (jreback) [reply pandas/tests/tslibs/test_conversion.py]: pls do this

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/tslibs/test_conversion.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    @pytest.mark.parametrize('tz', ['UTC', 'Asia/Tokyo',
+                                    'US/Eastern', 'Europe/Moscow'])
+    @pytest.mark.parametrize('freq', ['D', 'A'])
+    def test_tz_convert_single_matches_tz_convert(self, tz, freq):
+        tz_didx = date_range('2000-01-01', '2020-01-01', freq=freq, tz=tz)
+        utc_didx = date_range('2000-01-01', '2020-01-01', freq=freq)
+        compare_utc_to_local(tz_didx, utc_didx)
+        compare_local_to_utc(tz_didx, utc_didx)
+
+    def test_tz_convert_empty(self):
```
Comment (jreback) [inline pandas/tests/tslibs/test_conversion.py]: let's parameterize empty/nat and make a single test

File: pandas/tests/tslibs/test_timezones.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+import pytest
+import pytz
+import dateutil.tz
+
+from pandas._libs.tslibs import timezones
+from pandas import Timestamp
+
+
+class TestTimeZoneCacheKey(object):
```
Comment (jreback) [inline pandas/tests/tslibs/test_timezones.py]: no reason to make this part of a class

File: pandas/tests/tslibs/test_conversion.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+    @pytest.mark.parametrize('tz', ['UTC', 'Asia/Tokyo',
+                                    'US/Eastern', 'Europe/Moscow'])
+    def test_tz_convert_single_matches_tz_convert_hourly(self, tz):
+        # US: 2014-03-09 - 2014-11-11
+        # MOSCOW: 2014-10-26  /  2014-12-31
+        tz_didx = date_range('2014-03-01', '2015-01-10', freq='H', tz=tz)
+        utc_didx = date_range('2014-03-01', '2015-01-10', freq='H')
+        compare_utc_to_local(tz_didx, utc_didx)
+        # local tz to UTC can be differ in hourly (or higher) freqs because
```
Comment (jreback) [inline pandas/tests/tslibs/test_conversion.py]: blank line before comment

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/tseries/test_timezones.py [hunk 5]
```diff
@@ -2,15 +2,10 @@
...
 
 import pytz
-import dateutil
-import numpy as np
 
 from datetime import datetime
 
-import pandas.util.testing as tm
-from pandas.core.indexes.datetimes import date_range
-from pandas._libs import tslib
```
Comment (jreback) [inline pandas/tests/tseries/test_timezones.py]: what's left in this file?

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks

File: pandas/tests/tseries/test_timezones.py (hunk 5 above)
Comment (jbrockmendel) [reply pandas/tests/tseries/test_timezones.py]: Not much.  I'll try to kill it off after clearing out the current PR backlog.

**Pull request merged**